modular_skyrat/modules/modular_implants/code/soulcatcher/soulcatcher_component.dm

```diff
@@ -0,0 +1,273 @@
+		new_soul.soul_desc = "[new_soul] lacks a discernible form."
+
+	mind_to_add.transfer_to(new_soul, TRUE)
+	current_souls += new_soul
+	new_soul.current_room = WEAKREF(src)
+
+	var/datum/preferences/preferences = new_soul.client?.prefs
+	if(preferences)
+		new_soul.ooc_notes = preferences.read_preference(/datum/preference/text/ooc_notes)
+		if(!new_soul.body_scan_needed)
+			new_soul.soul_desc = preferences.read_preference(/datum/preference/text/flavor_text)
+
+	to_chat(new_soul, span_cyan("You find yourself now inside of: [name]"))
+	to_chat(new_soul, span_notice(room_description))
+	to_chat(new_soul, span_doyourjobidiot("You have entered a soulcatcher, do not share any information you have received while a ghost. If you have died within the round, you do not know your identity until your body has been scanned, standard blackout policy also applies."))
+	log_admin("[new_soul] entered the soulcatcher room, [src]")
+
+	return TRUE
+
+/// Removes a soul from a soulcatcher room, leaving it as a ghost.
+/datum/soulcatcher_room/proc/remove_soul(mob/living/soulcatcher_soul/soul_to_remove)
+	if(!soul_to_remove || !(soul_to_remove in current_souls))
+		return FALSE
+
+	qdel(soul_to_remove)
+	return TRUE
+
+/// Transfers a soul from a soulcatcher room to another soulcatcher room.
+/datum/soulcatcher_room/proc/transfer_soul(mob/living/soulcatcher_soul/target_soul, datum/soulcatcher_room/target_room)
+	if(!(target_soul in current_souls) || !target_room)
+		return FALSE
+
+	var/datum/component/soulcatcher/target_master_soulcatcher = target_room.master_soulcatcher.resolve()
+	if(target_master_soulcatcher != master_soulcatcher.resolve())
+		target_soul.forceMove(target_master_soulcatcher.parent)
+
+	target_soul.current_room = WEAKREF(target_room)
+	current_souls -= target_soul
+	target_room.current_souls += target_soul
+
+	to_chat(target_soul, span_cyan("you've been transfered to [target_room]!"))
+	to_chat(target_soul, span_notice(target_room.room_description))
+
+	return TRUE
+
+/**
+ * Sends a message or emote to all of the souls currently located inside of the soulcatcher room.
+ *
+ * Arguments
+ * * message_to_send - The message we want to send to the occupants of the room
+ * * message_sender - The person that is sending the message. This is not required.
+ * * emote - Is the message sent an emote or not?
+ */
+
+/datum/soulcatcher_room/proc/send_message(message_to_send, message_sender, emote = FALSE)
+	if(!message_to_send) //Why say nothing?
+		return FALSE
+
+	var/sender_name = ""
+	if(message_sender)
+		sender_name = "[message_sender] "
+
+	var/datum/asset/spritesheet/sheet = get_asset_datum(/datum/asset/spritesheet/chat)
+	var/tag = sheet.icon_tag("nif-soulcatcher")
+	var/soulcatcher_icon = ""
+
+	if(tag)
+		soulcatcher_icon = tag
+
+	var/message = ""
+	if(!emote)
+		message = "<font color=[SOULCATCHER_SAY_COLOR]>\ [soulcatcher_icon] <b>[sender_name]</b>says, \"[message_to_send]\"</font>"
```

Comment:
Why is this handled here and not in the CSS files? You shouldn't be doing UI stuff like that in DM.

(Basically it should be a span_soulcatcher_say() or something like that)

Reply:
I plan on having it so that the chat color can be changed in a later PR similar to hivemind, that is why I kept that here.